rsocket/statemachine/RSocketStateMachine.cpp

```diff
@@ -681,15 +706,18 @@ void RSocketStateMachine::handleUnknownStream(
   // TODO: comparing string versions is odd because from version
   // 10.0 the lexicographic comparison doesn't work
   // we should change the version to struct
-  if (frameSerializer_->protocolVersion() > ProtocolVersion{0, 0} &&
-      !streamsFactory_.registerNewPeerStreamId(streamId)) {
-    return;
+  if (frameType != FrameType::PAYLOAD) {
```

Comment:
Am I reading the code right that it is allowing PAYLOAD frame to come first on a new stream?

Reply:
No, the payload would be some other frame. The first frame received will be a `REQUEST_` type frame, however subsequent frames will be `PAYLOAD` frames in case of a fragmented request. These subsequent frames would fail the `!registerNewPeerStreamId` check. So the check is skipped if the frame is a `PAYLOAD`.